### DISCUSSION TITLE
using PICO as HID Keyboard

### DISCUSSION THREAD

**Discussion by Sweetkubuni** [micropython/micropython#15381] [RP2040 / Pico]
I am trying to build a simple keypad; which is based on the following example code : [keyboard example](https://github.com/micropython/micropython-lib/blob/master/micropython/usb/examples/device/keyboard_example.py). I did grab the following firmware version:  v1.24.0-preview.83.g0dd25a369. does this not support usb yet? I am getting no module named 'usb'

**Comment by mattytrentini** [comment]
`machine.USBDevice` should be installed but you'll also need to add the appropriate higher-level [usb libraries from micropython-lib](https://github.com/micropython/micropython-lib/tree/master/micropython/usb).

Damien gave a [talk on USBDevice](https://www.youtube.com/watch?v=ZIlrPTjIdSY) a couple of months ago and, just last week, gave a talk on how to implement a keyboard with the higher-level usb modules; the video isn't up yet but should be in a couple of days. I'll share it as soon as it's available!

It's still a new feature and there's not yet _a heap_ of examples or documentation.

**Comment by Gadgetoid** [comment]
We shipped a USB build of Pimoroni MicroPython (for Pico, not W) with the core USB libraries baked in, specifically for this purpose: https://github.com/pimoroni/pimoroni-pico/releases/download/v1.23.0-1/pico_usb-v1.23.0-1-pimoroni-micropython.uf2

I used this when writing the firmware for my whacky keyboard project: https://github.com/isoxliis/firmware-micropython/blob/main/isoxliis.py

I want to get Keybow 2040 up and running with similar code, too, though these aren't very exciting examples.

⚠️ the linked build uses a 1MB 1MB Flash/Filesystem split so *it will erase your .py files*. Back up before flashing 😬
